fhirpipe_clean/extract/mapping.py

```diff
@@ -0,0 +1,118 @@
+import json
+import os
+import requests
+
+def get_resources(source_name, from_file=None):
+    """
+    Get all available resources from a pyrog mapping.
+    The mapping may either come from a static file or from
+    a pyrog graphql API.
+
+    Args:
+        source_name: name of the project (eg: Mimic)
+        from_file (optional): path to the static file to mock
+                              the pyrog API response.
+    """
+    if from_file:
+        path = from_file
+        if not os.path.isabs(path):
+            path = os.path.join(os.getcwd(), path)
+
+        with open(path) as json_file:
+            resources = json.load(json_file)
+        source_json = resources["data"]["database"]
+
+        return source_json["resources"]
+
+    else:
+        # Get Source id from Source name
+        source = run_graphql_query(source_info_query, variables={"sourceName": source_name})
+        source_id = source["data"]["sourceInfo"]["id"]
+
+        # Check that Resource exists for given Source
+        available_resources = run_graphql_query(
+            available_resources_query, variables={"sourceId": source_id}
+        )
+        return available_resources["data"]["availableResources"]
+
+def prune_fhir_resource(resource_structure):
+    """ Remove FHIR attributes that have not been mapped
+    from resource structure object.
+    """
+    resource_structure["attributes"][:] = [attr for attr in resource_structure["attributes"] if rec_prune_resource(attr)]
```

Comment:
Interesting, what does `[:]` do?

Reply:
It's here to prevent the creation of a new array: the computation is made in place